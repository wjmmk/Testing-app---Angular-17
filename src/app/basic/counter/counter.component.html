<h1>Counters App!!!</h1>

<h3> Counter {{ counter }} </h3>
<hr>

<button class="btn btn-primary mx-1" (click)="increaseBy(+1)"> +1 </button>
<button class="btn btn-primary mx-1" (click)="increaseBy(-1)"> -1 </button>
