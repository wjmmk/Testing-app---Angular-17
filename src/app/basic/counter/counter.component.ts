import { Component, OnInit } from '@angular/core';

@Component({
  selector: 'app-counter',
  standalone: true,
  imports: [],
  templateUrl: './counter.component.html',
  styleUrl: './counter.component.scss'
})
export class CounterComponent implements OnInit {
 counter: number = 10;

 constructor() {}

 ngOnInit(): void {

 }

 increaseBy( value: number ) {
  this.counter += value
  //TODO:
  // console.log({ newValue: this.counter })
 }
}
