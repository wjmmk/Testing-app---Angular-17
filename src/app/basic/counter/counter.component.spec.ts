import { ComponentFixture, TestBed } from '@angular/core/testing';

import { CounterComponent } from './counter.component';

describe('CounterComponent', () => {
  let component: CounterComponent;
  let fixture: ComponentFixture<CounterComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [CounterComponent]
    })
    .compileComponents();

    fixture = TestBed.createComponent(CounterComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  test('should create', () => {
    expect(component).toBeTruthy();
  });

  test('It should be the same as the initial snapshot of the component', () => {
    expect(compiled).toMatchSnapshot();
  })

  test('should increase based on the arguments passed to the increaseBy(arg...)', () => {
    component.increaseBy(5);
    expect(component.counter).toBe(15)
  })

  test('It should increase the value when you press the +1 button', () => {
    const button = compiled.querySelectorAll('button')
    button[0].click()
    expect(component.counter).toBe(11)
  })

  test('I should update the value of the h3 tag related to the counter', () => {
    component.increaseBy(10)
    fixture.detectChanges()

    const h3 = compiled.querySelector('h3')
    expect(h3?.textContent).toContain('20')
  })
});
