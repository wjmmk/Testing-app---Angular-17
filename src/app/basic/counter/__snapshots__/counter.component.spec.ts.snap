// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterComponent It should be the same as the initial snapshot of the component 1`] = `
<div
  id="root1"
>
  <h1>
    Counters App!!!
  </h1>
  <h3>
     Counter 10 
  </h3>
  <hr />
  <button
    class="btn btn-primary mx-1"
  >
     +1 
  </button>
  <button
    class="btn btn-primary mx-1"
  >
     -1 
  </button>
</div>
`;
